pci/src/vfio.rs

```diff
@@ -320,17 +337,127 @@ impl VfioPciDevice {
             vfio_pci_configuration,
             mmio_regions: Vec::new(),
             interrupt: Interrupt {
+                intx: None,
                 msi: None,
                 msix: None,
             },
             mem,
         };
 
-        vfio_pci_device.parse_capabilities(interrupt_manager);
+        vfio_pci_device.parse_capabilities(msi_interrupt_manager);
+
+        vfio_pci_device.initialize_legacy_interrupt(legacy_interrupt_group)?;
 
         Ok(vfio_pci_device)
     }
 
+    fn enable_intx(&mut self) -> Result<()> {
+        if let Some(intx) = &mut self.interrupt.intx {
+            if !intx.enabled {
+                if let Some(eventfd) = intx.interrupt_source_group.notifier(0) {
+                    if let Err(e) = self
+                        .device
+                        .enable_irq(VFIO_PCI_INTX_IRQ_INDEX, vec![&eventfd])
+                    {
+                        warn!("Could not enable INTx: {}", e);
```

Comment:
Should there be an error propagated up here?

Reply:
Well it wasn't like that in the original code, but I think you're right, these calls shouldn't fail. Let me handle that in a follow up commit.